source/Irrlicht/CIrrDeviceSDL.cpp

```diff
@@ -218,9 +218,20 @@ int CIrrDeviceSDL::findCharToPassToIrrlicht(int assumedChar, EKEY_CODE key) {
 void CIrrDeviceSDL::resetReceiveTextInputEvents() {
 	gui::IGUIElement *elem = GUIEnvironment->getFocus();
 	if (elem && elem->acceptsIME())
+	{
 		SDL_StartTextInput();
+		core::rect<s32> pos = elem->getAbsolutePosition();
+		SDL_Rect rect;
+		rect.x = pos.UpperLeftCorner.X;
+		rect.y = pos.UpperLeftCorner.Y;
+		rect.w = pos.LowerRightCorner.X - rect.x;
+		rect.h = pos.LowerRightCorner.Y - rect.y;
+		SDL_SetTextInputRect(&rect);
```

Comment:
This should ideally be called before `SDL_StartTextInput` as per the [SDL docs](https://wiki.libsdl.org/SDL2/SDL_SetTextInputRect).

Reply:
Thanks. It seems like I went through the relevant SDL pages faster than I should.